audiomentations/augmentations/transforms.py

```diff
@@ -374,11 +390,31 @@ def randomize_parameters(self, samples, sample_rate):
     def apply(self, samples, sample_rate):
         num_places_to_shift = self.parameters["num_places_to_shift"]
         shifted_samples = np.roll(samples, num_places_to_shift, axis=-1)
+
+        if type(self.fade_duration) == float:
+            fade_length = int(sample_rate * self.fade_duration)
+        else:
+            fade_length = self.fade_duration
+
         if not self.rollover:
             if num_places_to_shift > 0:
                 shifted_samples[..., :num_places_to_shift] = 0.0
+                if self.fade:
+
+                    # When fading in, the first sample is always 0, that's why
+                    # the fade_length - 1 below.
+
+                    shifted_samples[
+                        ...,
+                        num_places_to_shift - 1 : num_places_to_shift + fade_length - 1,
```

Comment:
Can you check (with a unit test) if this breaks when fade_length is very large?

Reply:
I did in my new commit.